### PR TITLE
Add tests for dictionary comprehension

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR adds test cases for dictionary comprehension.

### CODE REVIEW

File: crates/ruff_python_parser/resources/invalid/expressions/dict/double_star_comprehension.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Double star means that the parser will try to parse it as a dictionary expression but
+# it's actually a comprehension.
+
+{**x: y for x, y in data}
+
+# TODO(dhruvmanila): This test case fails because there's no way to represent `**y`
+# in the AST. The parser tries to parse it as a binary expression but the range isn't
+# correct.
+# {x: **y for x, y in data}
```
Comment (dhruvmanila) [inline crates/ruff_python_parser/resources/invalid/expressions/dict/double_star_comprehension.py:9]: It's more like there's no way to represent `**y` if it's used in a key-value pair. A standalone double star expression is represented by adding `None` to keys vector and the expression to the `value` vector. The `None` in the corresponding position of the key vector signals that it's a double star expression.

Comment (MichaReiser) [reply crates/ruff_python_parser/resources/invalid/expressions/dict/double_star_comprehension.py:9]: I think we could recover here by parsing this as a power op without a left hand side:

```py
{ x: <missing>**y for x, y in data}
```

See https://github.com/microsoft/TypeScript/blob/02bb3108ad625cddcec228846b9ff56cc5398976/src/compiler/parser.ts#L5506-L5569

Comment (dhruvmanila) [reply crates/ruff_python_parser/resources/invalid/expressions/dict/double_star_comprehension.py:9]: Thanks for the reference. I'll make the change.

Comment (MichaReiser) [reply crates/ruff_python_parser/resources/invalid/expressions/dict/double_star_comprehension.py:9]: I should have been more explicit. I think that's an improvement we can make as a follow up PR or in the future. I don't see it as a priority (for as long as we track this somewhere)

**Pull request merged**